### PR TITLE
Add Prefect executor

### PR DESCRIPTION
**Pull request by rabernat**
Thanks to three the nice Executor examples @shoyer already wrote, I was able to write a Prefect Executor in less than an hour. It seems to work!

### CODE REVIEW

File: rechunker/executors/prefect.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+def _make_flow(specs: Iterable[StagedCopySpec]) -> prefect.Flow:
+    with prefect.Flow("Rechunker") as flow:
+        # iterate over different arrays in the group
+        for staged_copy_spec in specs:
+            copy_tasks = []
+            # iterate over the different stages of the array copying
+            for source, target, chunks in staged_copy_spec.stages:
+                keys = list(chunk_keys(source.shape, chunks))
```
Comment (shoyer) [inline rechunker/executors/prefect.py:45]: Would it make sense to make `chunk_keys` also a Prefect task? (That's basically what I did in BeamExecutor.)

Comment (rabernat) [reply rechunker/executors/prefect.py:45]: You mean a task that returns a list, which is then mapped from within the flow? Yes, I think that would work.

Comment (shoyer) [reply rechunker/executors/prefect.py:45]: I have no idea if it would actually perform, it seems like an obvious alternative :)

**Comment by TomAugspurger** [comment]
Updated for the new data model. @shoyer can you glance at https://github.com/pangeo-data/rechunker/pull/37/commits/2d2dcb8b0cce485287312671a28269cffc96e134 and let me know if it looks reasonable?

@rabernat you'll probably want to test this again after my changes.

File: tests/test_rechunk.py [hunk 2]
```diff
@@ -37,7 +38,7 @@ def target_chunks(request):
 @pytest.mark.parametrize("dtype", ["f4"])
 @pytest.mark.parametrize("max_mem", [25600000, "25.6MB"])
 @pytest.mark.parametrize(
-    "executor", ["dask", "python", requires_beam("beam")],
+    "executor", ["dask", "python", requires_beam("beam"), requires_prefect("prefect")],
```
Comment (shoyer) [inline tests/test_rechunk.py:41]: I would suggest also adding prefect to the group test case.

**Comment by TomAugspurger** [comment]
I think the coverage failure can be ignored.

**Pull request merged**